js/core/utils/data.js

```diff
@@ -191,7 +193,11 @@ export const toComparable = function(value, caseSensitive) {
     }
 
     if(!caseSensitive && typeof value === 'string') {
-        return value.toLowerCase();
+        if(options?.collateOptions?.sensitivity === 'base') {
+            value = value.normalize('NFD').replace(/[\u0300-\u036f]/g, '');
```

Comment:
As far as I understand, we remove here special unicode symbols

Let's move this RegExp to separate constant
E.g 
```typescript
const REMOVE_DIACRITICAL_MARKS_REGEXP = /[\u0300-\u036f]/g;
```